docs/subnets/create-a-vm-blobvm.md

```diff
@@ -0,0 +1,454 @@
+
+The VM exposes 3 things for interaction -
+* **Static and Non-Static Handlers (API)** for making requests to VM methods
+* **Service** for implementing handlers or APIs
+* **Client** for making calls to the service's APIs
+
+### What are Handlers
+
+Handlers serve the response for the incoming HTTP requests. Handlers can also be wrapped with **gRPC** for efficiently making calls from other services such as avalanchego. Handler helps in creating APIs. VM implements 2 kinds of handlers -
+
+* **Non-Static Handlers** - They help in interacting with blockchains instantiated by the VM. The API's endpoint will be different for different chains. `/ext/bc/[chainID]`
+* **Static Handlers** - They help in directly accessing VM. These are optional for reasons such as parsing genesis bytes required to instantiate new blockchains. `/ext/vm/[vmID]`
+
+### Registering a VM
+
+While registering a VM, we store its **factory** against the user-defined **vmID**. As the name suggests, a VM factory can create new VM instances. Using the VM's instance, we can initialize it and can create a new chain. Note that, instantiating a VM is different than initializing it. We initialize the instance of a VM with parameters like genesis, database manager, etc. to create a functional chain.
+
+```go
+// Registering a factory inside avalanchego
+func (m *manager) RegisterFactory(vmID ids.ID, factory Factory) error {
+    ...
+    m.factories[vmID] = factory
+    ...
+}
+```
+
+### Instantiating a VM
+
+Each VM has a **factory** that is capable of creating new VM instances. A VM can be initialized as a chain along with handlers for accessing it, only when we have the VM's instance. The factory's `New` method shown below, returns the VM's instance to its caller in `avalanchego`.
+
+```go
+// Returning a new VM instance from VM's factory
+func (f *Factory) New(*snow.Context) (interface{}, error) { return &vm.VM{}, nil }
+```
+
+VM's methods and other parts of the code are made available to `avalanchego` by **registering** its factory and instantiating a new VM using this factory whenever a chain is required to be built. It's generally in the [`factory.go`](https://github.com/ava-labs/blobvm/blob/master/factory.go) file of the VM.
+
+```go
+// Instantiating a new VM from its factory in avalanchego
+func (m *manager) buildChain(chainParams ChainParameters, sb Subnet) (*chain, error) {
+    ...
+    vmFactory, err := m.VMManager.GetFactory(vmID)
+    vm, err := vmFactory.New(ctx.Context)
+    ...
+}
+```
+
+### Initializing a VM
+
+Chains are functional, only when the instantiated VMs are initialized. Initializing a VM involves setting up the database, block builder, mempool, genesis state, handlers, etc. This will expose the VM's API handlers, start accepting transactions, gossiping them across the network, building blocks, etc.
```

Comment:
1. I would prefer to not use "etc" since it implies that the user has some pre-existing knowledge of how VM's work. If they're reading the tutorial, we shouldn't make that assumption.
2. I think it would be more helpful to just explain what each parameter in `VM#Initialize` does/is used for.

Reply:
Added more details at the bottom.